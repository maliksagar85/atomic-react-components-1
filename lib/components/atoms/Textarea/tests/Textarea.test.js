import React from 'react';
import { shallow, render } from 'enzyme';

import { TextareaVanilla } from '../index';

describe('<Textarea />', () => {
  let TextareaComponent = '';
  const sampleProps = {
    id: 'textarea-1',
    name: 'my-text-area',
    elementLocator: 'my-custom-textarea-selector',
  };
  beforeEach(() => {
    TextareaComponent = shallow(<TextareaVanilla {...sampleProps} />);
  });

  test('should render correctly', () => {
    expect(TextareaComponent).toMatchSnapshot();
  });

  test('should check the data attribute of textarea', () => {
    const inputVanilla = render(<TextareaVanilla {...sampleProps} />);

    expect(inputVanilla.attr('data-locator')).toEqual(sampleProps.elementLocator);
  });
});
