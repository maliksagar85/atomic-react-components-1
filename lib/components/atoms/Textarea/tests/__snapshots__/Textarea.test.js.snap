// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Textarea /> should render correctly 1`] = `
<textarea
  className=""
  data-locator="my-custom-textarea-selector"
  defaultValue=""
  id="textarea-1"
  name="my-text-area"
  placeholder=""
  resize="none"
  title=""
/>
`;
